blocks/algorithms/__init__.py

```diff
@@ -222,6 +224,29 @@ def compute_step(self, param, grad_wr_param):
         """
         raise NotImplementedError
 
+    def compute_steps(self, gradients):
+        """Build a Theano expression for steps for all parameters.
+
+        Override this method if you want to process the gradient
+        with respect to all parameters as a whole, not parameter-wise.
+
+        Parameters
+        ----------
+        gradients : dict of (:class:`~tensor.TensorSharedVariable`
```

Comment:
Hmm, yes, you're right. However, subsequent refactorings (or different clients of this code) might miss this subtlety! As such I would recommend either a) a comment around L189 stressing the importance of the deterministic iteration order or b) future-proofing by just using `OrderedDict`.